application/client/src/app/service/session/dependencies/search/charts/store.ts

```diff
@@ -18,20 +27,20 @@ export class ChartsStore extends Store<ChartRequest> {
         this.update([request as StoredEntity<ChartRequest>]);
     }
 
-    public tryRestore(smth: DisableConvertable): boolean {
-        if (smth instanceof ChartRequest) {
-            this.update([smth as StoredEntity<ChartRequest>]);
-            return true;
-        } else {
-            return false;
-        }
-    }
-
     public getActiveCount(): number {
         return this.get().filter((request) => request.definition.active).length;
     }
 
     public has(request: ChartRequest): boolean {
         return this.get().find((entity) => entity.isSame(request)) !== undefined;
     }
+
+    public select(event: ISelectEvent) {
+        this._selectedGuid = event.guids[0];
+        this.chartSelected.emit(event.entity === undefined ? '' : event.guids[0]);
+    }
+
+    public get selectedGuid(): string {
+        return this._selectedGuid;
+    }
```

Comment:
I guess this is UI actions, but it's a `store`, it's just storing information about session's charts. These methods related to state of target UI compenent.